src/lib/utils/check_table_equal.cpp

```diff
@@ -206,6 +214,16 @@ std::optional<std::string> check_table_equal(const std::shared_ptr<const Table>&
       print_table_comparison(error_type, error_msg, {{1, column_id}});
       return stream.str();
     }
+
+    if (ignore_nullable == IgnoreNullable::No && left_column_is_nullable != right_column_is_nullable) {
+      const std::string error_type = "Column NULLable mismatch (column " + std::to_string(column_id) + ")";
+      const std::string error_msg = std::string{"Actual column is "} + (left_column_is_nullable ? "" : "NOT ") +
+                                    "NULL\n" + std::string{"Expected column is "} +
+                                    (right_column_is_nullable ? "" : "NOT ") + "NULL";
+
+      print_table_comparison(error_type, error_msg, {{2, column_id}});
```

Comment:
Not sure, is the `2` a leftover and should also be replaced by `HEADER_SIZE`?

Reply:
Negative. It is row #2 that should be compared. If you look in lines 204 and 214, you will see 0 and 1, respectively. If this was HEADER_SIZE, we would always compare the last header row, even if adding another row means that the nullable row is no longer the last one.